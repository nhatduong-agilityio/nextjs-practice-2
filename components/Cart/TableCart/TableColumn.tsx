// Libs
import { Text, Th } from '@chakra-ui/react'
import React, { memo } from 'react'

interface IProps {
  label: string
  width?: string
  height?: string
}

const TableColumn = memo(({ label, width = 'full', height = '40px' }: IProps) => (
  <Th
    height={height}
    p={0}
    w={width}
    border='unset'
    textTransform='capitalize'
    alignItems='center'
    lineHeight='normal'
  >
    <Text size='25' variant='medium' color='#29293E' title='columnLabel'>
      {label}
    </Text>
  </Th>
))

TableColumn.displayName = 'TableColumn'

export default TableColumn
