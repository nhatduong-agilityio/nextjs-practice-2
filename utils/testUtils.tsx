import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
