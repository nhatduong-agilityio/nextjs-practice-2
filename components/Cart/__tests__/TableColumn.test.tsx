// Utils
import { render, screen } from '@utils/testUtils'

// Components
import { Table, Thead, Tr } from '@chakra-ui/react'
import TableColumn from '../TableCart/TableColumn'

describe('TableColumn render', () => {
  const propsMock = {
    label: 'Price',
  }

  it('Should TableColumn match Snapshot', () => {
    const { container } = render(
      <Table>
        <Thead>
          <Tr>
            <TableColumn {...propsMock} />
          </Tr>
        </Thead>
      </Table>,
    )

    expect(container).toMatchSnapshot()
  })

  it('Should render column label match Snapshot', () => {
    render(
      <Table>
        <Thead>
          <Tr>
            <TableColumn {...propsMock} />
          </Tr>
        </Thead>
      </Table>,
    )

    const label = screen.getByTitle('columnLabel')

    expect(label).toBeInTheDocument()
  })
})
