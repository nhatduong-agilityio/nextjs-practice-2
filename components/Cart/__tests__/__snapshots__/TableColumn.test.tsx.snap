// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableColumn render Should TableColumn match Snapshot 1`] = `
<div>
  <table
    class="chakra-table css-5605sr"
  >
    <thead
      class="css-0"
    >
      <tr
        class="css-0"
      >
        <th
          class="css-1xgyvoy"
        >
          <p
            class="chakra-text css-1x05orc"
            title="columnLabel"
          >
            Price
          </p>
        </th>
      </tr>
    </thead>
  </table>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
